### ISSUE TITLE
Learning Rate Schedulers' default dictionary parameters should be set via the Trainer

### ISSUE THREAD

**Issue by danieltudosiu** [PyTorchLightning/pytorch-lightning#1157]
## 🚀 Feature
The default Learning Rate Schedulers (LRS) dictionary parameters should be settable from the Trainer constructor. 

### Motivation
The documentation doesn't seem to be clear that the LRS have the following additional parameters available to be set when you configure the optimizers: 
```
    'interval': 'epoch',  # default every epoch
    'frequency': 1,  # default every epoch/batch
    'reduce_on_plateau': False,  # most often not ReduceLROnPlateau scheduler
    'monitor': 'val_loss'
```
### Pitch
Set those defaults in the constructor of the Trainer and the user can set them themselves. 

### Alternatives
Force the return type of the function LightningModule.configure_optimizers() to be List[Optimizer], List[Dict]

**Comment by SkafteNicki** [comment]
I disagree mainly due to:
* The trainer constructor already have many arguments, and this would potentially add 3 extra (the `'reduce_on_plateau'` is automatically determined from the LRS) arguments
* The user can already set these in `configure_optimizers` method by passing a LRS as a dict with these keywords.
* If I have two LRS and I want one to step after every epoch and one after every batch i.e. setting `interval="epoch"` for the first and `interval="step"` for the second, I would still need to change at least one of them in `configure_optimizers()` away from the default

I do agree that it is not very clear in the documentation that these are an option to set (this feature was introduced in 0.7.0), this probably needs to be changed
